src/psyclone/domain/common/algorithm/psyir.py

```diff
@@ -132,43 +171,48 @@ def _def_routine_root_name(self):
         :rtype: str
 
         '''
-        routine_root_name = "invoke_{0}".format(self._index)
-        if len(self.children) == 1:
-            # Add the name of the kernel if there is only one call
-            routine_root_name += "_" + self.children[0].name
+        if self._description:
+            routine_root_name = self._description.lower().strip()
+            if routine_root_name[0] == '"' and routine_root_name[-1] == '"' \
+               or \
+               routine_root_name[0] == "'" and routine_root_name[-1] == "'":
+                # fparser2 (issue #295) currently includes quotes as
```

Comment:
Thinking about this it may be a more general issues than the fparser double quotes. The character set of a Python string (UTF-8) can contain many characters that are not valid in a variable name. Maybe we need something like: `routine_root_name = routine_root_name.encode("ascii", "ignore").decode("utf-8")` (although this may be python 3 specific)

Also this may be an issue that needs to be solved in the Symbol or SymbolTable classes for any possible case.

Reply:
That is a good point. The old version does a check so I've replicated that.